src/dns_mirage/dns_resolver_mirage.mli

```diff
@@ -52,4 +52,4 @@ type static_dns =
 
 module Static : S with type stack = static_dns
 
-module Make(Time:Mirage_time.S)(S:Tcpip.Stack.V4) : S with type stack = S.t
+module Make(Time:Mirage_time.S)(S:Tcpip.Stack.V4V6) : S with type stack = S.t
```

Comment:
FWIW the Time parameter can be abandoned and all calls replaced by Mirage_sleep.ns... no need to depend on mirage-time (neither on mirage-clock) anymore.

Reply:
Indeed! Done in https://github.com/moby/vpnkit/pull/646/commits/1a4f62ae0243413d861af97854c0f55ea91fe0f9